numerics/sin_cos.cpp

```diff
@@ -33,6 +41,36 @@ static const __m128d sign_bit =
     _mm_castsi128_pd(_mm_cvtsi64_si128(0x8000'0000'0000'0000));
 }  // namespace masks
 
+template<FMAPolicy fma_policy>
+double FusedMultiplyAdd(double const a, double const b, double const c) {
+  if ((fma_policy == FMAPolicy::Force && CanEmitFMAInstructions) ||
+      (fma_policy == FMAPolicy::Auto && UseHardwareFMA)) {
+    using quantities::_elementary_functions::FusedMultiplyAdd;
+    return FusedMultiplyAdd(a, b, c);
+  } else {
+    return a * b + c;
+  }
+}
+
+void Reduce(Argument const x,
+            DoublePrecision<Argument>& x_reduced,
+            std::int64_t& quadrant) {
+  if (x < π / 4 && x > -π / 4) {
+    x_reduced.value = x;
+    x_reduced.error = 0;
+    quadrant = 0;
+  } else if (x <= π_over_2_threshold && x >= -π_over_2_threshold) {
+    std::int64_t const n = _mm_cvtsd_si64(_mm_set_sd(x * (2 / π)));
+    double const n_double = static_cast<double>(n);
```

Comment:
Is this faster than computing it using _mm_floor_sd?

Reply:
It is actually significantly faster, shaving 7 cycles on `Sin` and 5 on `Cos`.  The code is rather ugly, though.  Updated the benchmark numbers.